src/plugins/analysis/users_and_passwords/code/password_file_analyzer.py

```diff
@@ -6,113 +6,158 @@
-            for entry in passwd_entries:
-                self.update_file_object(file_object, entry_gen_function(entry))
-
-    def _add_found_password_tag(self, file_object: FileObject, result: dict):
-        for password_entry in result:
-            if 'password' in result[password_entry]:
-                username = password_entry.split(':', 1)[0]
-                password = result[password_entry]['password']
-                self.add_analysis_tag(
-                    file_object, f'{username}_{password}', f'Password: {username}:{password}', TagColor.RED, True
-                )
-
-    def update_file_object(self, file_object: FileObject, result_entry: dict):
-        file_object.processed_analysis[self.NAME].update(result_entry)
-        file_object.processed_analysis[self.NAME]['summary'].extend(list(result_entry))
-        self._add_found_password_tag(file_object, result_entry)
-
-
-def generate_unix_entry(entry: bytes) -> dict:
+class CredentialResult(pydantic.BaseModel):
+    username: str = Field(description='The username.')
+    full_entry: str = Field(description='The full entry in unparsed form.')
+    type: str = Field(description='The type of credential (UNIX, htpasswd, etc.).')
+    # FixMe: update type hint when we drop support for python <3.10
+    password_hash: Optional[str] = Field(description='The password in hashed form.', default=None)
+    password: Optional[str] = Field(description='The password (if the hash was available and cracked).', default=None)
+    error: Optional[str] = Field(
+        description='Error message (if cracking the password hash was not successful).', default=None
+    )
+
+
+class AnalysisPlugin(AnalysisPluginV0, AnalysisBasePluginAdapterMixin):
+    class Schema(pydantic.BaseModel):
+        credentials: List[CredentialResult] = Field(description='The list of found credentials.')
+
+    def __init__(self):
+        super().__init__(
+            metadata=AnalysisPluginV0.MetaData(
+                name='users_and_passwords',
+                description=(
+                    'search for UNIX, httpd, and mosquitto password files, parse them and try to crack the passwords'
+                ),
+                version='1.0.0',
+                Schema=self.Schema,
+                mime_blacklist=MIME_BLACKLIST_NON_EXECUTABLE,
+            ),
+        )
+
+    def analyze(self, file_handle: FileIO, virtual_file_path: dict[str, list[str]], analyses: dict) -> Schema:
+        del virtual_file_path, analyses
+        file_contents = Path(file_handle.name).read_bytes()
+        credentials: list[CredentialResult] = []
+        for regex_list, entry_gen_function in [
+            (UNIX_REGEXES, generate_unix_entry),
+            (HTPASSWD_REGEXES, generate_htpasswd_entry),
+            (MOSQUITTO_REGEXES, generate_mosquitto_entry),
+        ]:
+            credentials.extend(find_password_entries(file_contents, regex_list, entry_gen_function))
```

Comment:
I personally find this less readable than not using a for loop. Also not using a for-loop is three lines less.

Reply:
The `find_password_entries` was also ugly and kinda useless, since it had only one line. I refactored it in a way that made more sense to me